tests/mbt/model/ccv.qnt

```diff
@@ -654,6 +685,141 @@ module ccv {
+            val consumerState = currentState.consumerStates.get(consumer)
+            val vscPacket = consumerState.receivedVscPackets.select(packet => packet.id == vscId)[0]
+            val packet = Slash({
+                validator: consumerNode,
+                id: vscId,
+                downtime: downtime,
+                sendingTime: consumerState.chainState.runningTimestamp,
+                timeoutTime: consumerState.chainState.runningTimestamp + CcvTimeout.get(consumer),
+                valPower: vscPacket.validatorSet.get(consumerNode)
+
+            })
+            if (downtime) {
+                Err("Downtime slashes are not yet implemented")
+            } else {
+                val newConsumerState = commitPacketOnConsumer(consumerState, packet)
+                Ok(currentState.with("consumerStates", currentState.consumerStates.set(consumer, newConsumerState)))
+            }
+        }
+    }
+
+    // receives a given packet on the provider. The arguments are the consumer chain that sent the packet, and the packet itself.
+    pure def recvPacketOnProvider(currentState: ProtocolState, sender: Chain, packet: Packet): Result = {
+        match packet {
+            | VscMatured(p) => recvVscMaturedPacketOnProvider(currentState, sender, p)
+            | Slash(p) => recvSlashPacketOnProvider(currentState, sender, p)
+            | _ => Err("Do not know how to handle this packet type")
+        }
+    }
+
+    // TODO: currently this only jails, but it should eventually also slash
+    pure def recvSlashPacketOnProvider(currentState: ProtocolState, sender: Chain, packet: SlashPacket): Result = {
+        // determine the jail duration for the validator:
+        // if the packet is a downtime slash, the jail duration is DowntimeJailDuration
+        // otherwise, the validator is jailed until the DoubleSignJailTime
+        val jailEndTime = if (packet.downtime) {
+            packet.sendingTime + DowntimeJailTime
+        } else {
+            packet.sendingTime + CcvTimeout.get(sender)
+        }
+        // map the validator to its address on the provider
+        // (in general, the validator is a consumer address and could be an assigned one)
+        val providerVal = currentState.providerState.consumerAddrToValidator
+            .get(sender)
+            // if there is no entry for that validator, there is no key assignment
+            .getOrElse(packet.validator, packet.validator)
+        
+        // TODO: implement jailing
+        // val newProviderState = currentState.providerState
+        //  .jailUntil(currentState.providerState, packet.validator, jailEndTime)
+        //  .slash(currentState.providerState, packet.validator, packet.valPower, slashFactorToBeDetermined)
+
+        // Ok(currentState.with("providerState", newProviderState))
+        Err("not implemented")
+    }
+
+    pure def recvVscMaturedPacketOnProvider(currentState: ProtocolState, sender: Chain, packet: VscMaturedPacket): Result = {
```

Comment:
moved from ccv_utils because the "sibling function" recvSlashPacketOnProvider needs to access a parameter from the module